components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1014,250 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
+
+        result.setLimit(validateLimit(limit));
+        result.setOffSet(validateOffset(offset));
+        result.setSortBy(validateSortBy(sortBy));
+        result.setSortOrder(validateSortOrder(sortOrder));
+        result.setFilter(filter);
+    }
+
+    /**
+     * Validate sortBy.
+     *
+     * @param sortBy sortBy attribute.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortBy(String sortBy) {
+
+        if (StringUtils.isBlank(sortBy)) {
+            if (log.isDebugEnabled()) {
+                log.debug("sortBy attribute is empty. Therefore we set the default sortBy attribute. sortBy" +
+                        IdPManagementConstants.DEFAULT_SORT_BY);
+            }
+            return IdPManagementConstants.DEFAULT_SORT_BY;
+        }
+        switch (sortBy) {
+            case IdPManagementConstants.IDP_NAME:
+                sortBy = IdPManagementConstants.NAME;
+                break;
+            case IdPManagementConstants.IDP_HOME_REALM_ID:
+                sortBy = IdPManagementConstants.HOME_REALM_ID;
+                break;
+            default:
+                sortBy = IdPManagementConstants.DEFAULT_SORT_BY;
+                if (log.isDebugEnabled()) {
+                    log.debug("sortBy attribute is incorrect. Therefore we set the default sortBy attribute. " +
+                            "sortBy: " + IdPManagementConstants.DEFAULT_SORT_BY);
+                }
+                break;
+        }
+        return sortBy;
+    }
+
+    /**
+     * Validate sortOrder.
+     *
+     * @param sortOrder sortOrder ASC/DESC.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortOrder(String sortOrder) {
+
+        if (StringUtils.isBlank(sortOrder)) {
+            sortOrder = IdPManagementConstants.DEFAULT_SORT_ORDER;
+            if (log.isDebugEnabled()) {
+                log.debug("sortOrder is empty. Therefore we set the default sortOrder value as ASC. SortOrder: " +
+                        sortOrder);
+            }
+        } else if (sortOrder.equals(IdPManagementConstants.DESC_SORT_ORDER)) {
+            sortOrder = IdPManagementConstants.DESC_SORT_ORDER;
+        } else if (sortOrder.equals(IdPManagementConstants.ASC_SORT_ORDER)) {
+            sortOrder = IdPManagementConstants.ASC_SORT_ORDER;
+        } else {
+            sortOrder = IdPManagementConstants.DEFAULT_SORT_ORDER;
+            if (log.isDebugEnabled()) {
+                log.debug("sortOrder is incorrect. Therefore we set the default sortOrder value as ASC. SortOrder: "
+                        + sortOrder);
+            }
+        }
+        return sortOrder;
+    }
+
+    /**
+     * Validate limit.
+     *
+     * @param limit given limit value.
+     * @return validated limit and offset value.
+     */
+    private int validateLimit(int limit) {
+
+        try {
+            String maximumItemsPerPagePropertyValue =
```

Comment:
try {
                String maximumItemsPerPagePropertyValue =
                        IdentityUtil.getProperty(IdPManagementConstants.MAXIMUM_ITEMS_PRE_PAGE_PROPERTY);
                int maximumItemsPerPage;
                if (StringUtils.isNotBlank(maximumItemsPerPagePropertyValue)) {
                     maximumItemsPerPage = Integer.parseInt(maximumItemsPerPagePropertyValue);
                }else{
                    maximumItemsPerPage = SOME CONSTNANT EQUAL TO 100
                }
                    
                if (limit > maximumItemsPerPage) {
                        if (log.isDebugEnabled()) {
                            log.debug("Given limit exceed the maximum limit. Therefore we get the default limit from " +
                                    "identity.xml. limit: " + maximumItemsPerPage);
                        }
                        limit = maximumItemsPerPage;
                }
                
            } catch (NumberFormatException e) {
                limit = SOME CONSTNANT EQUAL TO 100
                log.warn("-----.", e);
            }
            return limit;
        }

Reply:
This is done